### ISSUE TITLE
instance loss

### ISSUE THREAD

**Issue by li-js** [DavyNeven/fastSceneUnderstanding#8]
Hi @DavyNeven,

It seems there is no L_reg term in the loss function in the [loss](https://github.com/DavyNeven/fastSceneUnderstanding/blob/master/lossf/myInstanceLoss.lua). Does it affect the performance a lot?

And also there seems to be a redundant term torch.sum(prediction) * 0 in the [loss]( 
https://github.com/DavyNeven/fastSceneUnderstanding/blob/master/lossf/myInstanceLoss.lua#L89).

**Comment by DavyNeven** [comment]
No, we didn't notice much difference when training the three tasks together, so we left it out. 

The redundant term is there because autograd can complain if there happens to be an image without any instances. In that case, no gradients would be calculated. By adding this extra term, the default gradient will be zero.

**Comment by li-js** [comment]
Thanks for the reply. 

The normalization [here](https://github.com/DavyNeven/fastSceneUnderstanding/blob/master/lossf/myInstanceLoss.lua#L82):
loss_dist = loss_dist + loss_d / ((#means - 1) + 1e-8) 
also seems inconsistant with the paper, as it idicates a coeffecient of 2/(C(C-1)). Any reasons?

**Comment by DavyNeven** [comment]
Indeed, the code is indeed slightly different than the formulas in the paper. In the paper, the loss is normalized per instance, so the variance term is divided by C and the distance loss is divided by C*(C-1)/2. In the code however, we don't normalize per instance. So this means that the variance term is not divided by C, and the distance loss is only divided by (C-1). Hope this is not to confusing.
